### PR TITLE
Closes cppl-56

### PR DESCRIPTION
**Pull request by iodream**
Implemented Signup page for client

### CODE REVIEW

File: src/Client/View/SignupPage/SignupPage.h [hunk 1]
```diff
@@ -14,6 +16,14 @@ class SignupPage : public QWidget
 	explicit SignupPage(QWidget *parent = nullptr);
 	~SignupPage();
 
+signals:
+	void Signup(SignupModel::SignupInDTO& dto);
+	void GotoLoginPage();
+
+private slots:
+	void on_signupSubmitButton_clicked();
```
Comment (MaksymTerekhovskyy) [inline src/Client/View/SignupPage/SignupPage.h:24]: Please, name the slots as a regular function (like OnSignupSubmitButtonClicked) and connect them to your buttons in constructor

File: src/Client/View/LoginPage/LoginPage.h [hunk 2]
```diff
@@ -21,11 +21,14 @@ class LoginPage : public QWidget
...
-
 signals:
 	void Login(LoginInDTO);
+	void GotoSignup();
 
 private slots:
 	void OnLoginSubmitButtonClicked();
+	void on_signupButton_clicked();
+	void on_passwordLineEdit_textChanged(const QString& arg1);
+	void on_loginLineEdit_textChanged(const QString& arg1);
```
Comment (MaksymTerekhovskyy) [inline src/Client/View/LoginPage/LoginPage.h:32]: Please also rename the slots there

**Review by MaksymTerekhovskyy — APPROVED** [review]

File: src/Client/Models/SignupModel.cpp [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+{
+	return (dto.password == dto.pass_repeat);
+}
+
+//checks data for presence of any characters at all
+bool SignupModel::CheckData(const SignupInDTO& dto) const
+{
+	auto username = dto.login;
+	auto pass = dto.password;
+	username.erase(std::remove_if(username.begin(), username.end(), ::isspace)
```
Comment (yaroslav-kernitskyi) [inline src/Client/Models/SignupModel.cpp:37]: This validator allows spaces inside login and password. Is this correct behaviour?

Comment (iodream) [reply src/Client/Models/SignupModel.cpp:37]: yes, i thought that was ok. it only checks if the field is whitespace-only

**Review by yaroslav-kernitskyi — APPROVED** [review]
Some indentation issues

**Pull request merged**